environment_test.go

```diff
@@ -53,6 +53,34 @@ func TestEnvironment_Basic(t *testing.T) {
 	env.chain.assertFailed(t)
 }
 
+type customReporter struct {
+	testing  *testing.T
+	reported bool
+}
```

Comment:
Small improvement idea: instead of creating a new struct `customReporter`, we can modify `mockReporter` as follows:

* add func() field to struct, say `reportCb func()`
* in Errorf, check if the field is non-nil, and if so, invoke it

Then in our test we can create mock reporter and set it's reportCb to a closure that invokes env.

Rationale:

* less code
* same reporter will be reused for deadlock tests for other structs too